sympy/tensor/indexed.py

```diff
@@ -137,6 +140,24 @@ class Indexed(Expr):
     is_symbol = True
     is_Atom = True
 
+    @staticmethod
+    def _filter_assumptions(kw_args):
+        """Split the given dict into two parts: assumptions and not assumptions.
+           Keys are taken as assumptions if they correspond to a handler on ``Q``."""
+        assumptions = {k: v for k, v in kw_args.items() if hasattr(Q, k)}
```

Comment:
If this is the same as Symbol, then shouldn't we use the assumptions that are used on Symbols?

I think those are `from sympy.core.assumptions import _assume_defined`. The only two there that are not in dir(Q) are 'polar' and 'noninteger'.

Reply:
That makes a lot of sense to me. I didn't find the `_assume_defined` set when I was looking for a way to do the filtering at first. See the latest commit.